src/core.js

```diff
@@ -695,19 +695,23 @@ function _api(api) {
 }
 
 function _apiPlugins(api) {
-  return api.plugins && api.ensure != 'removed' ? plugins(api.name, api.plugins) : [];
+  return api.plugins && !shouldBeRemoved(api) ? plugins(api.name, api.plugins) : [];
 }
 
 function _routePlugins(serviceName, route) {
-  return route.plugins && route.ensure != 'remove' ? routePlugins(serviceName, route.name, route.plugins) : [];
```

Comment:
The bug fixed in this commit is `'remove'` here as well as in L706 and L710. Those should all read `'removed'`. That check has been extracted into a function to prevent this bug from happening in the future. For anyone who was helping me debug, this was the source of most of my problems!